### PR TITLE
Initial commit for connector-threat-miner.

### PR DESCRIPTION
**Pull request by bidisha-spryiq**

### CODE REVIEW

File: threat-miner/info.json [hunk 1]
```diff
@@ -0,0 +1,278 @@
...
+            "output_schema": [
+                {
+                    "status_code": "",
+                    "status_message": "",
+                    "results": []
+                }
+            ],
+            "parameters": [
+                {
+                    "title": "Imphash",
```
Comment (cs-prerna-joshi) [inline threat-miner/info.json]:
```suggestion
                    "title": "Hash", or "Import Hash"
```

File: threat-miner/info.json [hunk 2]
```diff
@@ -0,0 +1,278 @@
...
+					]
+                }
+            ]
+        },
+		{
+            "title": "Get SSDeep",
+            "operation": "get_ssdeep_details",
+            "category": "investigation",
+            "description": "Retrieves the data that detect the level of similarity between two files at the binary level.",
+            "annotation": "ssdeep_details",
```
Comment (cs-prerna-joshi) [inline threat-miner/info.json]:
```suggestion
            "annotation": "get_ssdeep_details",
```

File: threat-miner/info.json [hunk 3]
```diff
@@ -0,0 +1,278 @@
...
+						"Report tagging"
+					]
+				}
+            ]
+        },
+        {
+            "title": "Get Samples",
+            "operation": "get_file_details",
+            "category": "investigation",
+            "description": "Returns known metadata of the given samples in the form of a note.",
```
Comment (cs-prerna-joshi) [inline threat-miner/info.json]:
```suggestion
            "description": "Returns known metadata of the given samples in the form of a note.",
```
is description correct?

File: threat-miner/info.json [hunk 4]
```diff
@@ -0,0 +1,278 @@
...
+						"Passive DNS",
+                        "Related Samples (hash only)",
+						"SSL Certificates (hash only)",
+						"Report tagging"
+					]
+				}
+            ]
+        },
+        {
+            "title": "Get Samples",
```
Comment (cs-prerna-joshi) [inline threat-miner/info.json]:
```suggestion
            "title": "Get File Details",
```

File: threat-miner/info.json [hunk 5]
```diff
@@ -0,0 +1,278 @@
...
+					"title": "Query Type",
+					"type": "select",
+					"name": "query_type",
+					"required": true,
+					"visible": true,
+					"editable": true,
+                    "description": "Select the type of query that is how you want to perform imphash operation.",
+                    "tooltip": "Select the type of query that is how you want to perform imphash operation.",
+					"options": [
+						"Samples"
```
Comment (cs-prerna-joshi) [inline threat-miner/info.json]: Do we need to add Report tagging option as well?

File: threat-miner/info.json [hunk 6]
```diff
@@ -0,0 +1,278 @@
...
+					"title": "Query Type",
+					"type": "select",
+					"name": "query_type",
+					"required": true,
+					"visible": true,
+					"editable": true,
+                    "description": "Select the type of query that is how you want to perform SSDeep operation.",
+                    "tooltip": "Select the type that is how you want to perform SSDeep operation.",
+					"options": [
+						"Samples"
```
Comment (cs-prerna-joshi) [inline threat-miner/info.json]: Do we need to add Report tagging option as well? and if there is only one option the no need for select type.

File: threat-miner/info.json [hunk 7]
```diff
@@ -0,0 +1,278 @@
...
+                    "editable": true,
+                    "value": "7bf5721bfa009479c33f3c3cf4ea5392200f030e",
+                    "placeholder": "e.g: 7bf5721bfa009479c33f3c3cf4ea5392200f030e",
+                    "description": "Required email address (SHA1 only) value to retrieved Whois records.",
+                    "tooltip": "Required email address (SHA1 only) value to retrieved Whois records."
+                }
+            ]
+        }
+	]
+}
```
Comment (cs-prerna-joshi) [inline threat-miner/info.json:252]: Please reformat the file

File: threat-miner/operations.py [hunk 8]
```diff
@@ -0,0 +1,128 @@
...
+        endpoint = '/v2/domain.php?q={0}&rt={1}'.format(params.get('domain_name'), rt)
+        return self.make_api_call(endpoint=endpoint)
+
+    def get_ip_details(self, params):
+        q_type = params.get('query_type')
+        if q_type == 'WHOIS':
+            rt = 1
+        elif q_type == 'Passive DNS':
+            rt = 2
+        elif q_type == 'Related Samples (Hash only)':
```
Comment (cs-prerna-joshi) [inline threat-miner/operations.py]:
```suggestion
        elif q_type == 'Related Samples (hash only)':
```

File: threat-miner/info.json (hunk 5 above)
Comment (bidisha-spryiq) [reply threat-miner/info.json]: The Report tagging API is not working. So, removed the select query type field. By default samples query types will be executed.

File: threat-miner/info.json (hunk 6 above)
Comment (bidisha-spryiq) [reply threat-miner/info.json]: The Report tagging API is not working. So, removed the select query type field. By default samples query types will be executed.

**Review by cs-prerna-joshi — APPROVED** [review]

**Pull request merged**